godtools/ViewControllers/Tract/TractViewController.swift

```diff
@@ -227,6 +231,23 @@ class TractViewController: BaseViewController {
     
     // MARK: - Handle page movements
     
+    func moveToPage(notification: Notification) {
+        guard let dictionary = notification.object as? [String: String] else {
+            return
+        }
+        
+        let destinationViewPageId = dictionary["pageId"]
+        for pageId in self.pagesIds.keys {
```

Comment:
I think you could do:
````swift
for (pageId, pageIndex) in self.pageIds {
..
}
````
so that you can already have the value instead of doing `self.pagesIds[pageId]!` . One of the small nice things in Swift!